### ISSUE TITLE
Pricing individual books & build issue

### ISSUE THREAD

**Issue by surelynot** [Simbul/baker#544]
I've been taking a look at the latest Master fork of Baker, after messing about with Andrew's shelf version for a while – glad to see these coming together.

I've spent the morning setting up a test application in iTunes Connect with a paid product (the 'A Study In Scarlet' listed on the example books.json) and a free subscription to do some testing. 

To get the entries in the books.json file synced with the products in the 'In App Purchases' section, am I correct in assuming that it would require having the 'name' field (in books.json) and the 'reference name field (in iTunes Connect) matching? Does this automatically enter the price next to the item on the shelf in the user interface?
eg.
http://simon-maidment.com/ScreenShot-2.png
http://simon-maidment.com/ScreenShot-3.png

When I've tried to test this today by adding the required info to constants.h and uploading the book and books.json file to a server, but am getting these errors on attempting to build in Xcode:
http://simon-maidment.com/ScreenShot-1.png

Hope someone can help. My hope is to eventually use this framework for two apps, one that is the vehicle for a journal, with subscriptions and using newsstand, the other for books, with no subscriptions and individual prices for each, not using the newsstand, more just a shelf with in app purchases. Thanks in advance. Simon.

**Comment by Simbul** [comment]
Hello.

At the moment, Baker only supports free subscriptions. You're correct in thinking that, eventually, the "name" and "reference name" fields will have to match, but right now Baker does not contain any logic to display prices or talk to the App Store to complete the purchase.

Also, at the moment Baker only supports two modes:
1. Newsstand-based shelf with downloads and a free subscription (and eventually individual prices and paid subscriptions)
2. Local shelf (no Newsstand) with books bundled in the application, no downloads or purchases

The first feature will definitely be implemented eventually, the second might be, but it may take some time.

As for the compilation issue, which version of Xcode are you using? `UIInterfaceOrientationMaskAll` was introduced in iOS 6, so you may need to update to a newer version.

**Comment by surelynot** [comment]
Hi Simbul,

thanks, version 6.0 of Xcode builds it ok – my laptop hadn't been updated.

I'm sure people are hard at work with the IAP development. There's a couple of early tutorials that @viggiosoft has previously posted, which may be some use for people working on this (I know Andrew et al had been looking at them earlier in the year while working on their shelf fork):

http://www.viggiosoft.com/blog/blog/2011/10/29/at-newsstand-and-subscriptions/
https://github.com/viggiosoft/HowToMakeAMagazineApp
http://ios-blog.co.uk/tutorials/how-to-make-a-magazine-app-in-ios-part-i/
http://ios-blog.co.uk/tutorials/how-to-make-a-magazine-app-in-ios-part-ii/

Cheers!

**Comment by nin9creative** [comment]
We got as far as implementing the subscriptions (1 year, 6 months, 3 months, free) etc...  I'm sure that 
there is some code that can be harvested from the shelf branch we made for helping with some of this.

We never got around to implementing the single issue ad hoc purchase.

At this point, I'm committed to working as much as I can to speed up the Baker 4.0+ releases as our other
shelf work was trying to get something done for folks chomping at the bit to have a shelf for Baker.

I actually like the implementation in the 4.0 stream better than our previous work :)  Plus the work that
the Baker crew did on refactoring the Baker Reader to be more modular is a heck of a lot better than the
hackjob I did on the prior 3.X version to get it work with the other shelf fork :)
